user_interaction_txt_text.py

```diff
@@ -0,0 +1,94 @@
+import os
+from enum import Enum, auto
+from typing import Optional
+
+from user_file_manager import UserFileManager
+
+
+class MessageScope(Enum):
+    GLOBAL = auto()
+    ACTIVE = auto()
+    PASSIVE = auto()
+
+
+class UserInteraction():
+    def __init__(self, context: 'Context'):
+        self.context = context
+        self.user_names = self.context.user_names
+
+        self.file_manager = UserFileManager(
+            os.path.join(os.path.dirname(__file__), "History"))
+        self.game = self.context._game
```

Comment:
Don\t fetch protected fields from context.
This add strong dependencies in the code (in case if GameController implementation will change it will be needed to change code in UserInteraction)

Please, use context public fields

Reply:
Done